ui/src/app/shared/components/input-filter.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import {Autocomplete} from '../../../../node_modules/argo-ui';
+import {ErrorPanel} from './error-panel';
+
+interface InputProps {
+    value: string;
+    placeholder: string;
+    type: string;
+    onChange: (input: string) => void;
+}
+
+interface InputState {
+    input: string;
+    localInputs: string[];
+    error?: Error;
+}
+
+export class InputFilter extends React.Component<InputProps, InputState> {
```

Comment:
This is abstracted from original `NamespaceFilter` to make it a generic input box with local cache.